Sources/Nimble/Matchers/SatisfyAllOf.swift

```diff
@@ -35,6 +36,7 @@ internal func satisfyAllOf<T>(_ predicates: [Predicate<T>]) -> Predicate<T> {
     }
 }
 
+@available(*, deprecated)
 public func && <T>(left: Predicate<T>, right: Predicate<T>) -> Predicate<T> {
```

Comment:
All the arguments and return type are `Predicate` so should not be deprecated.

Reply:
This means the inner implementation shouldn't use `Matcher`, in that case